### PR TITLE
8265421: java/lang/String/StringRepeat.java test is missing a memory requirement

### PR DESCRIPTION
**Pull request by mychris**
Please review the following patch, which fixes the mentioned test case for memory constrained devices. This can be tested on a linux based development machine, using systemd as follows:

```
$ systemd-run --user --scope -p MemoryMax=800M -p MemorySwapMax=0 /usr/bin/make TEST="test/jdk/java/lang/String/StringRepeat.java" run-test
```

I split up the test case in a part which only executes the small repeat counts, and in one part which executes the huge repeat count. The second part is guarded by a requirement, so it is only executed if enough memory is available.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8265421](https://bugs.openjdk.java.net/browse/JDK-8265421): java/lang/String/StringRepeat.java test is missing a memory requirement

### Reviewers
 * [Jim Laskey](https://openjdk.java.net/census#jlaskey) (@JimLaskey - **Reviewer**)
 * [Aleksey Shipilev](https://openjdk.java.net/census#shade) (@shipilev - **Reviewer**)
 * [Rahul Yadav](https://openjdk.java.net/census#ryadav) (@rhyadav - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/3566/head:pull/3566` \
`$ git checkout pull/3566`

Update a local copy of the PR: \
`$ git checkout pull/3566` \
`$ git pull https://git.openjdk.java.net/jdk pull/3566/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 3566`

View PR using the GUI difftool: \
`$ git pr show -t 3566`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/3566.diff">https://git.openjdk.java.net/jdk/pull/3566.diff</a>

</details>

### CODE REVIEW

**Review by JimLaskey — APPROVED** [review]

File: test/jdk/java/lang/String/StringRepeat.java [hunk 1]
```diff
@@ -27,21 +27,33 @@
  * @run main/othervm -Xmx2g StringRepeat
```
Comment (shipilev) [inline test/jdk/java/lang/String/StringRepeat.java:27]: So, does `-Xmx2g` make sense in this configuration now?

Comment (mychris) [reply test/jdk/java/lang/String/StringRepeat.java:27]: No, not really. The first test case instance should require way less than 1G of memory, so I will just remove the heap configuration for that instance.

**Review by JimLaskey — APPROVED** [review]

**Review by shipilev — APPROVED** [review]

**Review by rhyadav — APPROVED** [review]

**Comment by mychris** [comment]
Thanks a lot for the reviews.

/integrate

**Pull request closed without merging**